### PR TITLE
Add documentation for installing components from user-defined URLs

### PR DESCRIPTION
**Pull request by tomekpapiernik**
**Description**

Changes proposed in this pull request:

- Added "Configuration" doc describing installing Kyma components from user-defined URLs
- Updated Installation CR description

**Related issue(s)**
#6512

### CODE REVIEW

File: docs/kyma/05-04-component-source-url.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+---
+title: Install components from user-defined URLs
+type: Configuration
+---
+
```
Comment (piotrmsc) [inline docs/kyma/05-04-component-source-url.md]: Please describe what will happen if : 
- I specify remote component which is at the same time a component compiled in kyma-installer like istio
- I specify component like above and fetching the remote chart will fail, will the compiled in be used?
- Describe the retry mechanism, does it differ to the current one?
- Would be nice to describe basic authentication for HTTP. See here https://github.com/hashicorp/go-getter#basicauthentication for GIT we can skip as it requires the key to be present in the container which cannot work.

File: docs/kyma/05-04-component-source-url.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+The Kyma Operator allows you to use external URLs as sources for the components you decide to install Kyma with. Using this mechanism, you can install Kyma with one customized component, which you, for example, modified and store in GitHub or as a `.zip` or `.tgz` archive on a server, and have other components using the officially released sources.
+
+To install a component using an external URL as the source, you must add the **source.url** attribute to the entry of a component in the Installation custom resource (CR).
+
+The address must expose the `chart.yaml` of the component directly. This means that for Git repositories or archives that do not store this file at the top level, you must specify the path to the file.
```
Comment (piotrmsc) [inline docs/kyma/05-04-component-source-url.md]: If you have `chart.yaml` helm will complain with an error : ```Error: chart metadata (Chart.yaml) missing```
```suggestion
The address must expose the `Chart.yaml` of the component directly. This means that for Git repositories or archives that do not store this file at the top level, you must specify the path to the file.
```

File: docs/kyma/05-04-component-source-url.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+To specify the exact location of the `chart.yaml`, append it to the URL beginning with two backslashes `//` to indicate the path within the archive or repository. See these sample entries for components with user-defined source URLs from the Installation CR for more details:
```
Comment (piotrmsc) [inline docs/kyma/05-04-component-source-url.md]:
```suggestion
To specify the exact location of the `Chart.yaml`, append it to the URL beginning with two backslashes `//` to indicate the path within the archive or repository. See these sample entries for components with user-defined source URLs from the Installation CR for more details:
```

File: docs/kyma/05-04-component-source-url.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+</div>
+
+>**NOTE:** Read [this](#custom-resource-installation) document to learn more about the Installation CR.
+
+### Error handling and retry policy
+
+If you specify an external URL as a source for a Kyma component, the Kyma Operator attempts to access it 3 times during the installation process. If it fails to reach the specified URL in one of the 3 attempts or fails to find the required files, the installation step fails and the entire installation process starts over.
+
+There is no fallback mechanism implemented. This means that in a case where the Operator fails to install a component using a custom URL, the installation step always fails, even if the component sources are included in the Kyma Installer image.
```
Comment (Tomasz-Smelcerz-SAP) [inline docs/kyma/05-04-component-source-url.md]:
```suggestion
There is no fallback mechanism implemented. This means that in a case where the Operator fails to install a component using a custom URL, the installation step always fails, even if the default component sources are included in the Kyma Operator image.
```

File: docs/kyma/05-04-component-source-url.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
+---
+title: Install components from user-defined URLs
+type: Configuration
+---
+
+The Kyma Operator allows you to use external URLs as sources for the components you decide to install Kyma with. Using this mechanism, you can install Kyma with one customized component, which you, for example, modified and store in GitHub or as a `.zip` or `.tgz` archive on a server, and have other components using the officially released sources.
```
Comment (klaudiagrz) [inline docs/kyma/05-04-component-source-url.md]: I have the impression that this sentence is too long and thus confusing. Can we split it into chunks to make it more digestive?

File: docs/kyma/05-04-component-source-url.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+To install a component using an external URL as the source, you must add the **source.url** attribute to the entry of a component in the Installation custom resource (CR).
```
Comment (klaudiagrz) [inline docs/kyma/05-04-component-source-url.md]: Link?
https://kyma-project.io/docs/#custom-resource-installation

File: docs/kyma/05-04-component-source-url.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+  - Archive with `Chart.yaml` at the top level:
+    ```
+    - name: "ory"
+      namespace: "kyma-system"
+      source:
+        url: https://hosting.com/your-user/files/kyma-custom-ory.zip
+    ```
+
+
+  - Archive with `Chart.yaml` deeper in file structure:
```
Comment (klaudiagrz) [inline docs/kyma/05-04-component-source-url.md]:
```suggestion
  - Archive with `Chart.yaml` deeper in the file structure:
```

File: docs/kyma/05-04-component-source-url.md [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+
+  - Repository with `Chart.yaml` at the top level:
+    ```
+    - name: "cluster-essentials"
+      namespace: "kyma-system"
+      source:
+        url: github.com/my-project/kyma.git
+    ```
+
+  - Repository with `Chart.yaml` deeper in file structure:
```
Comment (klaudiagrz) [inline docs/kyma/05-04-component-source-url.md]:
```suggestion
  - Repository with `Chart.yaml` deeper in the file structure:
```

File: docs/kyma/05-04-component-source-url.md [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+      namespace: "kyma-system"
+      source:
+        url: github.com/my-project/kyma.git//resources/cluster-essentials
+    ```
+
+  </details>
+
+</div>
+
+>**NOTE:** Read [this](#custom-resource-installation) document to learn more about the Installation CR.
```
Comment (klaudiagrz) [inline docs/kyma/05-04-component-source-url.md]: Consider linking to the Installation CR in the second paragraph when you first mention it.

File: docs/kyma/05-04-component-source-url.md [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+
+### Error handling and retry policy
+
+If you specify an external URL as a source for a Kyma component, the Kyma Operator attempts to access it 3 times during the installation process. If it fails to reach the specified URL in one of the 3 attempts or fails to find the required files, the installation step fails and the entire installation process starts over.
```
Comment (klaudiagrz) [inline docs/kyma/05-04-component-source-url.md]:
```suggestion
If you specify an external URL as a source for a Kyma component, the Kyma Operator attempts to access it three times during the installation process. If it fails to reach the specified URL in one of the three attempts or fails to find the required files, the installation step fails and the entire installation process starts over.
```

**Pull request merged**